### ISSUE TITLE
Disallowed clients aren't checked at all when "Allowed clients" is configured

### ISSUE THREAD

**Issue by skreza** [AdguardTeam/AdGuardHome#2007]
### Issue Details

* **Version of AdGuard Home server:**
  v0.103.3
* **How did you setup DNS configuration:**
  Upstream to cloudflare via https and tls
* **If it's a router or IoT, please write device model:**
  Running docker version
* **Operating system and version:**
  Latest Docker build from docker hub running on Ubuntu 20.04 server

### Expected Behavior
On the main page you have a list of the Top client, hovering over a client shows a 'Block' button.
Clicking the block button generates a confirmation box once confirmed the client turns to a shade of red and an 'Unblock' button is now visible.

Expected result would be that the client would effectivily be unable to resolve any DNS via AdGuard.

### Actual Behavior
Clients marked as blocked can still resolve DNS via AdGuard. I've cleared the DNS cache on the client yet it can still resolve dns no issue. Client only has 1 DNS server configured which is AdGuard and tests with dig and nslookup also resolve without issue from AdGuard. I would have assumed this should have caused the client to no longer be able to resolve any DNS requests from AdGuard?

**Comment by szolin** [comment]
What do you see in "DNS settings" -> "Disallowed clients" list?

**Comment by ameshkov** [comment]
@szolin could it be that we apply Disallowed clients to plain DNS only?

**Comment by skreza** [comment]
Ok I think I've figured it out, under the 'Allowed clients' list I have the CIDR for my internal subnet to ensure AdGuard only accepts requests from this address range. It appears that Allowed has priority over Disallowed, I removed the CIDR and the block started applying.

Not sure if this should be the case though, generally any explicit deny has highest priority.

**Comment by KizunaH** [comment]
https://github.com/AdguardTeam/AdGuardHome/issues/2150

**Comment by ameshkov** [comment]
Merged into https://github.com/AdguardTeam/AdGuardHome/issues/2075
